lib/charms/layer/kubernetes_common.py

```diff
@@ -933,3 +934,199 @@ def get_secret_password(username):
             if username == secret_user:
                 return password
     return None
+
+
+def get_node_ip():
+    """Determines the preferred NodeIP value for this node."""
+    cidr = cluster_cidr()
+    if not cidr:
+        return None
+    if is_ipv6_preferred(cidr):
+        return get_ingress_address6("kube-control")
+    else:
+        return get_ingress_address("kube-control")
```

Comment:
Removed [here](https://github.com/charmed-kubernetes/charm-kubernetes-worker/pull/103/files#diff-1837fd4ad476d5c87c628d68b3c461fc5c03d6aeafc928e910c69aa6ae267f85L520-L530) and added in this PR